File: uts_pbo_smt3/pertemuan_3/tugas_3.py
class Mahasiswa:
    def __init__(self):
        self._nama = ""
        self._nim = ""

    def set_nama(self, nama):
        self._nama = nama

    def get_nama(self):
        return self._nama

    def set_nim(self, nim):
        if len(nim) == 10 and nim.isdigit():
            self._nim = nim
        else:
            print("NIM harus 10 digit angka.")

    def get_nim(self):
        return self._nim


# Contoh penggunaan
m = Mahasiswa()
m.set_nama("Budi")
m.set_nim("2023123456")

print("Nama:", m.get_nama())
print("NIM :", m.get_nim())

File: uts_pbo_smt3/pertemuan_3/tugas_3.java
public class Mahasiswa {
    private String nama;
    private String nim;

    // Setter nama
    public void setNama(String nama) {
        this.nama = nama;
    }

    // Getter nama
    public String getNama() {
        return this.nama;
    }

    // Setter nim (dengan validasi 10 digit)
    public void setNim(String nim) {
        if (nim.length() == 10 && nim.matches("[0-9]+")) {
            this.nim = nim;
        } else {
            System.out.println("NIM harus 10 digit angka.");
        }
    }

    // Getter nim
    public String getNim() {
        return this.nim;
    }

    public static void main(String[] args) {
        Mahasiswa m = new Mahasiswa();
        m.setNama("Andi");
        m.setNim("2023123456"); // valid

        System.out.println("Nama: " + m.getNama());
        System.out.println("NIM: " + m.getNim());
    }
}
